src/openassetio-core/include/openassetio/hostApi/ManagerFactory.hpp

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include <openassetio/export.h>
+#include <openassetio/typedefs.hpp>
+
+OPENASSETIO_FWD_DECLARE(hostApi, HostInterface)
+OPENASSETIO_FWD_DECLARE(hostApi, ManagerInterfaceFactoryInterface)
+OPENASSETIO_FWD_DECLARE(managerApi, ManagerInterface)
+OPENASSETIO_FWD_DECLARE(LoggerInterface)
+
+namespace openassetio {
+inline namespace OPENASSETIO_CORE_ABI_VERSION {
+namespace hostApi {
+
+OPENASSETIO_DECLARE_PTR(ManagerFactory)
+
+/**
+ * The ManagerFactory is the primary mechanism for querying for
+ * available @ref manager "managers" and constructing a
+ * @fqref{hostApi.Manager} "Manager", using the supplied manager
+ * interface factory.
```

Comment:
[fix] Could we add a comment along the lines of "Hosts should never attempt to directly construct a `Manager` class or interact with the implementation factory directly" - just to really spell it out...

Reply:
Added your suggestion in 4a07d906